docs/observability.md

```diff
@@ -12,6 +12,101 @@ If you need to troubleshoot the Collector, see [Troubleshooting].
 Read on to learn about experimental features and the project's overall vision
 for internal telemetry.
 
+<!-- toc -->
+
+- [Goals of internal telemetry](#goals-of-internal-telemetry)
+  * [Observable elements](#observable-elements)
+  * [Impact](#impact)
+  * [Configurable level of observability](#configurable-level-of-observability)
+  * [Internal telemetry properties](#internal-telemetry-properties)
+    + [Units](#units)
+    + [Process for defining new metrics](#process-for-defining-new-metrics)
+- [Experimental trace telemetry](#experimental-trace-telemetry)
+
+<!-- tocstop -->
+
+## Goals of internal telemetry
+
+The Collector's internal telemetry is an important part of fulfilling
+OpenTelemetry's [project vision](vision.md). The following section explains the
+priorities for making the Collector an observable service.
+
+### Observable elements
+
+The following aspects of the Collector need to be observable.
+
+- [Current values]
+  - Some of the current values and rates might be calculated as derivatives of
+    cumulative values in the backend, so it's an open question whether to expose
+    them separately or not.
+- [Cumulative values]
+- [Trace or log events]
+  - For start or stop events, an appropriate hysteresis must be defined to avoid
+    generating too many events. Note that start and stop events can't be
+    detected in the backend simply as derivatives of current rates. The events
+    include additional data that is not present in the current value.
+- [Host metrics]
+  - Host metrics can help users determine if the observed problem in a service
+    is caused by a different process on the same host.
+
+### Impact
+
+The impact of these observability improvements on the core performance of the
+Collector must be assessed.
+
+### Configurable level of observability
+
+Some metrics and traces can be high volume and users might not always want to
+observe them. An observability verboseness “level” allows configuration of the
```

Comment:
Since we already use 'verbosity' for the configuration key, should we use it here as well?
```suggestion
observe them. An observability verbosity “level” allows configuration of the
```